common/persistence/visibility/store/elasticsearch/query_interceptors.go

```diff
@@ -166,7 +166,7 @@ func parseSystemSearchAttributeValues(name string, value any) (any, error) {
 		}
 	case searchattribute.ExecutionStatus:
 		if status, isNumber := value.(int64); isNumber {
-			value = enumspb.WorkflowExecutionStatus_name[int32(status)]
+			value = enumspb.WorkflowExecutionStatus(status).String()
```

Comment:
You should check whether it's a valid WorkflowExecutionStatus first before doing this, otherwise we'll query ES even when we know it's an invalid status (like 3134569)